cli/cmd/helm/upgrade.go

```diff
@@ -179,3 +188,187 @@ func upgradeTobs(cmd *cobra.Command, args []string) error {
+	if err != nil {
+		return fmt.Errorf("failed to get the secret existing secret during the upgrade %s: %v", existingDBSecret, err)
+	}
+
+	var admin, postgres, standby []byte
+
+	if bytepass, exists := s.Data["admin"]; exists {
+		admin = bytepass
+	} else {
+		return fmt.Errorf("could not get TimescaleDB password: %w", errors.New("user not found"))
+	}
+
+	if bytepass, exists := s.Data["postgres"]; exists {
+		postgres = bytepass
+	} else {
+		return fmt.Errorf("could not get TimescaleDB password: %w", errors.New("user not found"))
+	}
+
+	if bytepass, exists := s.Data["standby"]; exists {
+		standby = bytepass
+	} else {
+		return fmt.Errorf("could not get TimescaleDB password: %w", errors.New("user not found"))
+	}
+
+	sec := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      newDBsecret,
+			Namespace: root.Namespace,
+			Labels:    utils.GetTimescaleDBsecretLabels(),
+		},
+		Data: map[string][]byte{
+			"PATRONI_REPLICATION_PASSWORD": standby,
+			"PATRONI_admin_PASSWORD":       admin,
+			"PATRONI_SUPERUSER_PASSWORD":   postgres,
+		},
+		Type: "Opaque",
+	}
+
+	err = k8s.CreateSecret(sec)
+	if err != nil {
+		return fmt.Errorf("failed to create secret during the upgrade %s: %v", sec.Name, err)
+	}
+
+	fmt.Printf("secret/%s created\n", newDBsecret)
+
+	return nil
+}
+
+func copyDBCertificate() error {
+	fmt.Println("Migrating the certificate")
+	existingDBCertificate := root.HelmReleaseName + "-timescaledb-certificate"
+	newDBCertificate := root.HelmReleaseName + "-certificate"
+
+	s, err := k8s.KubeGetSecret(root.Namespace, existingDBCertificate)
+	if err != nil {
+		return fmt.Errorf("failed to get the secret existing secret during the upgrade %s: %v", existingDBCertificate, err)
+	}
+
+	sec := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      newDBCertificate,
+			Namespace: root.Namespace,
+			Labels:    utils.GetTimescaleDBsecretLabels(),
+		},
+		Data: s.Data,
+		Type: "Opaque",
+	}
+
+	err = k8s.CreateSecret(sec)
+	if err != nil {
+		return fmt.Errorf("failed to create secret during the upgrade %s: %v", sec.Name, err)
+	}
+
+	fmt.Printf("secret/%s created\n", newDBCertificate)
+
+	return nil
+}
+
+func copyDBBackup() error {
+	existingDBBackup := root.HelmReleaseName + "-timescaledb-pgbackrest"
+	newDBBackUp := root.HelmReleaseName + "-pgbackrest"
+
+	s, err := k8s.KubeGetSecret(root.Namespace, existingDBBackup)
+	if err != nil {
+		// backup is optional is disabled skip backup secret copying
+		ok := errors2.IsNotFound(err)
+		if !ok {
+			return fmt.Errorf("failed to get the secret existing secret during the upgrade %s: %v", existingDBBackup, err)
+		}
+		return nil
+	}
+
+	fmt.Println("Migrating backup configuration")
+
+	newData := make(map[string][]byte)
+	for key, value := range s.Data {
+		ok := strings.Contains(key, "-s3-")
+		if ok {
+			newKey := strings.Replace(key, "-", "_", -1)
+			newKey = strings.ToUpper(newKey)
+			newKey = "PGBACKREST_" + newKey
+			newData[newKey] = value
```

Comment:
should the value be base64 encoded?

Reply:
encoding is done internally. We just pass the data in the format of bytes.